python/taichi/lang/kernel_impl.py

```diff
@@ -800,7 +802,7 @@ def callback(original, updated):
             ret_dt = self.return_type
             has_ret = ret_dt is not None
 
-            if has_ret:
+            if has_ret or _check_in_IPython():
```

Comment:
Hi, this logic is added since in Jupyter's magic cell (`%%time` and `%%timeit`), when the kernel has a `print` but no return statement, the printed info will not be flushed to the screen when the cell is executed the first time but will be printed along with other cells when another cell is executed. Forcing the kernel to sync fixes this problem.
You can run the attached file to reproduce this problem,.
[taichi_jupyter_demo.zip](https://github.com/taichi-dev/taichi/files/9948789

Reply:
I see, I'd assume this bug existed with the current `sourceinspect` module as well right? In that case this should go into a separate PR with a proper IPython test with `%%time` in it?

Reply:
The existing bug with the current `sourceinspect` is that it would simply crash if we use cell magics in Jupyter.